### ISSUE TITLE
[UX Issue] Scrolling With Middle Click Speed is Too Slow

### ISSUE THREAD

**Issue by ahmadalli** [sumatrapdfreader/sumatrapdf#1062]

**Comment by SumatraPeter** [comment]
So the middle-click scrolling speed is slow only in Sumatra and not any other apps?

Mouse or touchpad? Which model? Latest drivers installed from manufacturer's site (not Windows Update)?

Is the behaviour the same with the latest [pre-release version](https://www.sumatrapdfreader.org/prerelease.html) (you can test a portable copy without affecting your current installed version)?

**Comment by ahmadalli** [comment]
By scrolling using middle click I mean press middle button then drag the mouse while still pressing the button.

**Comment by GitHubRulesOK** [comment]
My mouse (Logitech) in SumatraPDF 3.1.2 windows 10x64 1803 etc works at the same speed as whilst reviewing this page in Internet Explorer, what did have a bearing is this page at present is about 10 times shorter than the document I opened to compare, so my perception is it took longer in SumatraPDF to get from one end to the other, otherwise there was no noticeable difference in the scroll rate.

**Comment by SumatraPeter** [comment]
>By scrolling using middle click I mean pressing the middle button then dragging the mouse while still pressing the button.

Why do you have to drag while still holding the middle button down? Why not just middle-click so that the auto-scroll icon (![image](https://user-images.githubusercontent.com/15380332/44620577-78db0780-a8b4-11e8-938a-5697d6e2f6c2.png)) shows up, then simply release the button and move the mouse? _The farther the mouse is moved from where you middle-clicked, the faster the document will scroll in that direction._ This means if I middle-click at the very top of the page (i.e. right below Sumatra's toolbar), release the button and move the cursor all the way to the bottom of the page, PDFs scroll vertically at a pretty fast clip.

>This functionality is handled by softwares and is not driver related.

Depends. Drivers are software too, and they can override app-specific auto-scroll behavior. For example I have a mouse whose driver allows universal auto-scrolling to be enabled, thus allowing it to be used even in apps that don't ordinarily support the feature such as Windows Explorer for example.

**Comment by ahmadalli** [comment]
> Why do you have to drag while still holding the middle button down? Why not just middle-click so that the 

yeah that does the same thing :)

> Depends. Drivers are software too of course, and they can definitely override app-specific auto-scroll behavior. 

yeah but in compare to google chrome, Sumatra is too slow for me. And I thought maybe the default is too slow since the usage of middle click scrolling (with proper distance from click position) is to fast scroll pages but the longest distance was still too slow.

**Comment by GitHubRulesOK** [comment]
Just to confirm I switched to a Microsoft mouse with both vertical and horizontal (tilt) control, the only way I can slow it down is to zoom right in (at least 600%) otherwise at  normal zoom levels the page can generally scroll so fast lines become unreadable, both mice seems to be auto-scrolling fast enough here.
both this and Logitech use the default MS Driver Date: 06/21/2006 Driver Version: 10.0.17134.1

**Comment by ahmadalli** [comment]
this is my experience with google chrome:

![2018-08-26_10-21-14](https://user-images.githubusercontent.com/2167301/44625244-fdac3c80-a919-11e8-9d22-63ecc13456dd.gif)

and this is my experience with Sumatra: https://imgur.com/a/k8XCpzQ

@GitHubRulesOK I tried with recommended scaling (125%, my default is 100%) and disable windows messing with dpi but still the difference between chrome and sumatra is the same as the gifs.

**Comment by GitHubRulesOK** [comment]
A gif is worth a 1000 words

OK so it is important to find two apps that can open the same long file, 
I tried XPS but MS XPS viewer and other XPS viewer has NO autoscroll !!! (one up to SumatraPDF)

So resorted to a long PDF in Firefox preview and SumatraPDF, I have to agree FireFox is many times faster when autoscrolling, however I normally use autoscroll when skim reading (and SumatraPDF speed is really good for that) otherwise I use the scroll bar (or bookmarks) to "fast forwards"

Acrobat has configurable scroll speed 0-9 and some users complain they can not slow it down enough for reading i.e. music scores or legal documents etc.
"I used AutoScroll to read long documents, such as depositions. However, I am  unable to control the scrolling speed. In some documents, it scrolls slowly  enough to read, and I can slow it down more by moving the mouse. In others, it is too fast to read and cannot be slowed down"
" I find that the minimum speed of AutoScroll is too fast to read anything. As far as I know, there's no adjustment -- so I just  don't use it. Another nice idea poorly implemented. :-(" Microsoft Word MVP 

In my humble opinion the rate in SumatraPDF is perfect for a reader, and allows fine control when zoomed in to details, so I would not want it jittering at high speed. For me its fine the way it is and if not broken ….

**Comment by SumatraPeter** [comment]
I was gonna say your GIF and MP4 aren't comparable given the obvious disparity in document lengths, but GitHubRulesOK's already discussed that in detail. I don't use Chrome but don't disagree that it might allow much faster auto-scrolling than Sumatra.

So basically this boils down to **a feature request for configurable auto-scroll speed in Sumatra**, because simply speeding it up is not going to do for all those who find the current top speed more than adequate.

**Comment by GitHubRulesOK** [comment]
@ kjk this may be closed ? as @SumatraPeter has pointed out this is effectively a duplicate of User adjustable scrolling speed  #873

**Comment by SumatraPeter** [comment]
Hmm... Actually @GitHubRulesOK @kjk isn't #873 about scrolling sped while _rotating_ the mouse wheel, whereas this one is about the auto-scroll speed on _clicking_ the mouse wheel or middle button?

**Comment by GitHubRulesOK** [comment]
@SumatraPeter 
You are right that the two threads come from a different perspective (same as #557 ) however the underlying issue is the same "how to control scroll rate in one app" (independent of means) 
As I understand it and @kjk can correct me, SumatraPDF "responds" to an external instruction that may come from screen, touch pad, mouse, or joystick etc. so whatever the external cause is its the speed that is desired to be changed by users **for some faster for others slower**. this can be simply achieved by applying a divisor or multiplier in given steps like on the TV remote fast forward / slow mo  e.g x.25 x.5 x.75 x1 x1.5 x2 x4 (note that is an exponential range :-)
